prover/crates/lib/circuit_prover_service/src/metrics.rs

```diff
@@ -0,0 +1,46 @@
+use std::time::Duration;
+
+use vise::{Buckets, Histogram, Metrics};
+
+/// Metrics for witness vector generator execution
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "witness_vector_generator")]
+pub struct WitnessVectorGeneratorMetrics {
+    /// How long does it take to load witness vector inputs?
+    #[metrics(buckets = Buckets::LATENCIES)]
+    pub pick_time: Histogram<Duration>,
+    /// How long does it take to synthesize witness vector?
+    #[metrics(buckets = Buckets::LATENCIES)]
+    pub synthesize_time: Histogram<Duration>,
+    /// How long does it take to send witness vectors to gpu prover?
+    #[metrics(buckets = Buckets::LATENCIES)]
+    pub transfer_time: Histogram<Duration>,
+    /// How long does it take to save witness vector failure?
+    #[metrics(buckets = Buckets::LATENCIES)]
+    pub save_time: Histogram<Duration>,
+}
+
+#[vise::register]
+pub static WITNESS_VECTOR_GENERATOR_METRICS: vise::Global<WitnessVectorGeneratorMetrics> =
+    vise::Global::new();
+
+/// Metrics for GPU circuit prover execution
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "circuit_prover")]
+pub struct CircuitProverMetrics {
+    /// How long does it take to load prover inputs?
+    #[metrics(buckets = Buckets::LATENCIES)]
```

Comment:
nit: no need to resolve now, most likely we would need to do this across the codebase, but there is also a `unit` macro to determine unit of measurement, like seconds, minutes etc.